flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -105,6 +106,9 @@
 	/** The current or latest execution attempt of this vertex's task. */
 	private volatile Execution currentExecution;	// this field must never be null
 
+	/** input split*/
+	private ArrayList<InputSplit> inputSplits;
```

Comment:
It seems like we can initialize this on construction and make the field `final`, in particular as it is used with `synchronized`.

Reply:
Done.